sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryQuerySource.java

```diff
@@ -172,28 +159,43 @@ private synchronized JobStatistics dryRunQueryIfNeeded(BigQueryOptions bqOptions
   }
 
   private void executeQuery(
+      String jobIdToken,
       String executingProject,
-      String jobId,
       TableReference destinationTable,
       JobService jobService) throws IOException, InterruptedException {
+    // Generate a transient (random) query job ID, because this code may be retried after the
```

Comment:
Does this actually happen ? I thought query execution and table deletion steps are not fused.

Reply:
They both happen as part of BigQuerySourceBase.split() - they're not separate steps.